bundles/org.openhab.automation.pythonscripting/src/main/resources/lib/openhab/helper.py

```diff
@@ -84,16 +86,30 @@ def __init__(self, name=None, description=None, tags=None, triggers=None, condit
         self.conditions = conditions
         self.profile = profile
 
-    def __call__(self, clazz):
+    def __call__(self, clazzOrFunc):
         proxy = self
 
-        class_package = proxy.getClassPackage(clazz.__name__)
+        if isfunction(clazzOrFunc):
+            class_package = clazzOrFunc.__name__
+            _rule_obj = clazzOrFunc
+        else:
+            class_package = proxy.getClassPackage(clazzOrFunc.__name__)
+            _rule_obj = clazzOrFunc()
 
-        clazz.logger = Java_LogFactory.getLogger( "{}.{}".format(LOG_PREFIX, class_package) )
+        clazzOrFunc.logger = Java_LogFactory.getLogger( LOG_PREFIX + "." + file_package + "." + class_package )
```

Comment:
"file_package" is already part of the LOG_PREFIX. I changed it some days ago to support webUI based scripts, because there is no filename available